internal/common/debug/example_test.go

```diff
@@ -36,8 +38,54 @@ type (
 	stopperImpl struct {
 		pollerTracker *pollerTrackerImpl
 	}
+
+	// activityTrackerImpl implements the ActivityTracker interface
+	activityTrackerImpl struct {
+		sync.RWMutex
+		m map[ActivityInfo]int64
+	}
+
+	// activityStopperImpl implements the Stopper interface
+	activityStopperImpl struct {
+		sync.Once
+		info    ActivityInfo
+		tracker *activityTrackerImpl
+	}
 )
 
+var _ ActivityTracker = &activityTrackerImpl{}
+var _ Stopper = &activityStopperImpl{}
+
+func (ati *activityTrackerImpl) Start(info ActivityInfo) Stopper {
+	ati.Lock()
+	ati.m[info]++
+	ati.Unlock()
```

Comment:
Nit: can we defer this instead? I dont think line 61 will fail, but just in case